### PR TITLE
fix: update dataset card with descriptive category and paper title reference

### LINKED ISSUE #2807: Update dataset card template
### Description of the feature

https://huggingface.co/datasets/mteb/CodeFeedbackST/discussions/1#68484384790ce37286a91214

> [CoIR: A Comprehensive Benchmark for Code Information Retrieval Models](https://arxiv.org/abs/2407.02883)

> Would require us to parse the BibTeX to extract the title before auto-generating the sheet, but I think that is doable.
I also think the task category might be a bit misleading. Ideally, I would like it to be Retrieval (text-to-text) (t2t = text-to-text and the task.type is Retrieval)

### PR DESCRIPTION
**Pull request by fzoll**
## Summary
- Format task category as `{type} ({description})` (e.g. `Retrieval (text-to-text)`) instead of raw codes like `t2t`
- Extract paper title from BibTeX citation and format reference as a markdown link (e.g. `[CoIR: A Comprehensive Benchmark...](url)`) instead of a plain URL

**Before** (CodeFeedbackST example):
| | |
|---|---|
| Task category | t2t |
| Reference | https://arxiv.org/abs/2407.02883 |

**After**:
| | |
|---|---|
| Task category | Retrieval (text-to-text) |
| Reference | [CoIR: A Comprehensive Benchmark for Code Information Retrieval Models](https://arxiv.org/abs/2407.02883) |

## Test plan
- [x] Verified with CodeFeedbackST task — category renders as `Retrieval (text-to-text)`, reference renders as markdown link with paper title
- [x] Tested edge cases: no BibTeX, no reference URL, no category — all degrade gracefully
- [x] All existing tests pass (1232+ metadata validation tests, card tests, citation tests)

Closes #2807

### CODE REVIEW

**Comment by Samoed** [comment]
Seems that in this RP changes from https://github.com/embeddings-benchmark/mteb/pull/4216

**Comment by KennethEnevoldsen** [comment]
Yeah let us get the https://github.com/embeddings-benchmark/mteb/pull/4216 to avoid reviewing code twice

**Comment by fzoll** [comment]
@Samoed @KennethEnevoldsen Eh, my bad, mistakenly used the wrong base. I rebased; this should be clean now.

File: mteb/abstasks/task_metadata.py [hunk 1]
```diff
@@ -273,6 +274,30 @@
 21. v2t: video to text
 """
 
+CATEGORY_DESCRIPTIONS: dict[str, str] = {
```
Comment (Samoed) [inline mteb/abstasks/task_metadata.py]: We have mapping of valid categories in that will be applied https://github.com/embeddings-benchmark/mteb/blob/a29dc993d2aceb7e021b85e4fdb75455979bb7f9/mteb/abstasks/task_metadata.py#L827

File: mteb/abstasks/task_metadata.py [hunk 2]
```diff
@@ -623,20 +648,50 @@ def _create_dataset_card_data(
...
             # parameters for readme generation
             dict(
                 citation=self.bibtex_citation,
                 dataset_description=self.description,
-                dataset_reference=self.reference,
+                dataset_reference=formatted_reference,
                 descriptive_stats=descriptive_stats,
                 dataset_task_name=self.name,
-                category=self.category,
+                category=formatted_category,
```
Comment (Samoed) [inline mteb/abstasks/task_metadata.py:658]: This category only for table in 

Image

Probably we can remove it overall

File: mteb/abstasks/task_metadata.py [hunk 3]
```diff
@@ -623,20 +648,50 @@ def _create_dataset_card_data(
...
                 domains=", ".join(self.domains) if self.domains else None,
             ),
         )
 
+    @staticmethod
+    def _extract_bibtex_title(bibtex_citation: str) -> str | None:
```
Comment (Samoed) [inline mteb/abstasks/task_metadata.py:675]: I think it's better to move out from `metadata`

File: mteb/abstasks/task_metadata.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply mteb/abstasks/task_metadata.py:658]: I think it is better the proposed way no?

Comment (Samoed) [reply mteb/abstasks/task_metadata.py:658]: Agree, that's better, but this will dupulicate information from other dataset tags

File: mteb/abstasks/task_metadata.py (hunk 3 above)
Comment (KennethEnevoldsen) [reply mteb/abstasks/task_metadata.py:675]: I think it is quite decent here - though we could create a module for bibtex utilities (though that seems like a seperate issue)
